packages/firestore/src/core/event_manager.ts

```diff
@@ -59,16 +58,21 @@ export class EventManager implements SyncEngineListener {
     this.syncEngine.subscribe(this);
   }
 
-  listen(listener: QueryListener): Promise<TargetId> {
+  async listen(listener: QueryListener): Promise<void> {
     const query = listener.query;
     let firstListen = false;
 
     let queryInfo = this.queries.get(query);
     if (!queryInfo) {
       firstListen = true;
       queryInfo = new QueryListenersInfo();
-      this.queries.set(query, queryInfo);
     }
+
+    if (firstListen) {
+      queryInfo.viewSnap = await this.syncEngine.listen(query);
+    }
+
+    this.queries.set(query, queryInfo!);
```

Comment:
I'm surprised the `!` is required here. Isn't `queryInfo` guaranteed to be set here? (What confuses me is you're not null-checking the assignment to e.g. `queryInfo.viewSnap`).